### PR TITLE
add code to merge splited clusters in blech_post_process.py

### PR DESCRIPTION
**Pull request by jianyoulin**

### CODE REVIEW

File: blech_post_process.py [hunk 1]
```diff
@@ -145,18 +152,25 @@ class unit_descriptor(tables.IsDescription):
 		# Ask the user for the split clusters they want to choose
 		chosen_split = easygui.multchoicebox(msg = 'Which split cluster do you want to choose? Hit cancel to exit', choices = tuple([str(i) for i in range(n_clusters)]))
 		try:
-			chosen_split = int(chosen_split[0])
+			#chosen_split = int(chosen_split[0])
```
Comment (narendramukherjee) [inline blech_post_process.py]: Maybe remove the commented out old line?

File: blech_post_process.py [hunk 2]
```diff
@@ -172,25 +186,76 @@ class unit_descriptor(tables.IsDescription):
...
-			unit_description[unit_type[0]] = 1		
-		unit_description.append()
-		table.flush()
-		hf5.flush()
+		if split_merge:
+			unit_waveforms = []
+			unit_times = []
+			cluster_unit_waveforms = spike_waveforms[np.where(predictions == int(clusters[0]))[0], :]
+			cluster_unit_times = spike_times[np.where(predictions == int(clusters[0]))[0]]
+			for cluster_ in chosen_split:
```
Comment (narendramukherjee) [inline blech_post_process.py]: Let's name cluster_ as split_cluster ?

File: blech_post_process.py [hunk 3]
```diff
@@ -172,25 +186,76 @@ class unit_descriptor(tables.IsDescription):
...
+					unit_type = easygui.multchoicebox(msg = 'What type of unit is this (Regular spiking = Pyramidal cells, Fast spiking = PV+ interneurons)?', choices = ('regular_spiking', 'fast_spiking'))
+					unit_description[unit_type[0]] = 1
+				unit_description.append()
+				table.flush()
+				hf5.flush()
+			else:
+				continue
+
+		else:#if not split_merge:
+			hf5.create_group('/sorted_units', unit_name)
```
Comment (narendramukherjee) [inline blech_post_process.py:191]: We have to change this part to include each chosen_split as a separate unit. So we will have the same for loop as when `split_merge` is `True` : `for split_cluster in chosen_split` - now each `split_cluster` will be included as a separate unit.

Comment (narendramukherjee) [reply blech_post_process.py:191]: Code will look something like:
```
else:
	for split_cluster in chosen_split:
		hf5.create_group('/sorted_units', unit_name)
		unit_waveforms = spike_waveforms[np.where(predictions == int(clusters[0]))[0], :]	# Waveforms of originally chosen cluster
                unit_waveforms = unit_waveforms[np.where(split_predictions == chosen_split)[0], :]	# Subsetting this set of waveforms to include only the chosen split
                unit_times = spike_times[np.where(predictions == int(clusters[0]))[0]]			# Do the same thing for the spike times
                unit_times = unit_times[np.where(split_predictions == chosen_split)[0]]
                waveforms = hf5.create_array('/sorted_units/%s' % unit_name, 'waveforms', unit_waveforms)
                times = hf5.create_array('/sorted_units/%s' % unit_name, 'times', unit_times)
		unit_description['electrode_number'] = electrode_num
		single_unit = easygui.multchoicebox(msg = 'I am almost-SURE that electrode: %i cluster: %i split: %i is a beautiful single unit (True = Yes, False = No)' % (electrode_num, int(cluster), int(split_cluster)), choices = ('True', 'False'))
		unit_description['single_unit'] = int(ast.literal_eval(single_unit[0]))
		# If the user says that this is a single unit, ask them whether its regular or fast spiking
		unit_description['regular_spiking'] = 0
		unit_description['fast_spiking'] = 0
		if int(ast.literal_eval(single_unit[0])):
			unit_type = easygui.multchoicebox(msg = 'What type of unit is this (Regular spiking = Pyramidal cells, Fast spiking = PV+ interneurons)?', choices = ('regular_spiking', 'fast_spiking'))
			unit_description[unit_type[0]] = 1
		unit_description.append()
		table.flush()
		hf5.flush()				

		# Finally increment max_unit and create a new unit name
		max_unit += 1
		unit_name = 'unit%03d' % int(max_unit + 1)

		# Get a new unit_descriptor table row for this new unit
		unit_description = table.row
```

**Comment by jianyoulin** [comment]
Not sure if this pull request is specific to blech_post_process.py. If not, could you please only look at the post_processing code? Thank you.

**Review by narendramukherjee — APPROVED** [review]

**Pull request merged**